src/manager/evm.rs

```diff
@@ -26,127 +30,213 @@ pub type MiddlewareImpl = SignerMiddleware<Provider<Http>, Wallet<SigningKey>>;
+        let min_validator_stake = params
+            .min_validator_stake
+            .atto()
+            .to_u64()
+            .ok_or_else(|| anyhow!("invalid min validator stake"))?;
+
+        let params = IsubnetActorConstructorParams {
+            // TODO: replace this with parent
+            parent_id: ipc_registry::SubnetID {
+                route: agent_subnet_to_evm_addresses(&params.parent)?,
+            },
+            name: params.name,
+            ipc_gateway_addr: (*self.gateway_contract).address(),
+            consensus: params.consensus as u64 as u8,
+            min_activation_collateral: ethers::types::U256::from(min_validator_stake as u128),
+            min_validators: params.min_validators,
+            bottom_up_check_period: params.bottomup_check_period as u64,
+            top_down_check_period: params.topdown_check_period as u64,
+            // TODO: update this variable properly
+            majority_percentage: 50,
+            genesis: ethers::types::Bytes::default(),
+        };
+        log::info!("creating subnet: {params:?}");
+
+        let call = self.registry_contract.new_subnet_actor(params);
+        let pending_tx = call.send().await?;
+        let receipt = pending_tx.retries(10).await?;
+        return match receipt {
+            Some(r) => {
+                for log in r.logs {
+                    log::debug!("log: {log:?}");
+
+                    match ethers_contract::parse_log::<ipc_registry::SubnetDeployedFilter>(log) {
+                        Ok(subnet_deploy) => {
+                            let ipc_registry::SubnetDeployedFilter {
+                                subnet_addr,
+                                subnet_id,
+                            } = subnet_deploy;
+
+                            log::debug!("subnet with id {subnet_id:?} deployed at {subnet_addr:?}");
+
+                            // subnet_addr.to_string() returns a summary of the actual address, not
+                            // usable in the actual code.
+                            let subnet_addr = format!("{subnet_addr:?}");
+                            log::debug!("raw subnet addr: {subnet_addr:}");
+
+                            let eth_addr = EthAddress::from_str(&subnet_addr)?;
+                            return Ok(Address::from(eth_addr));
+                        }
+                        Err(_) => {
+                            log::debug!("not of event subnet actor deployed, continue");
+                            continue;
+                        }
+                    }
+                }
+                Err(anyhow!("no logs receipt"))
+            }
+            None => Err(anyhow!("no receipt to event, txn not successful")),
+        };
     }
 
     async fn join_subnet(
         &self,
         subnet: SubnetID,
-        from: Address,
+        _from: Address,
         collateral: TokenAmount,
-        params: JoinParams,
+        _params: JoinParams,
```

Comment:
There are no params here? Don't we need to pass the multiaddress for the validator as part of the join? If this is the case there's something wrong with the solidity contract

Reply:
Yeah, the solidity contract does not take the validator address as parameter. Let me check with them to verify this on the latest code.